internal/service/ivs/recording_configuration.go

```diff
@@ -0,0 +1,334 @@
+package ivs
+
+import (
+	"context"
+	"errors"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ivs"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func ResourceRecordingConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceRecordingConfigurationCreate,
+		ReadWithoutTimeout:   resourceRecordingConfigurationRead,
+		DeleteWithoutTimeout: resourceRecordingConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"destination_configuration": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"s3": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"bucket_name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z0-9-.]{3,63}$`), "must contain only lowercase alphanumeric characters, hyphen, or dot, and between 3 and 63 characters"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
```

Comment:
I kept going back and forth on this part of the schema, whether to have a single string for `bucket_name` or this nested structure. The nested structure closely mirrors the model used inside the [`CreateRecordingConfiguration` API call](https://docs.aws.amazon.com/ivs/latest/APIReference/API_CreateRecordingConfiguration.html). Is this nested structure okay?